mlops/common/pipeline_utils.py

```diff
@@ -0,0 +1,227 @@
+from mlops.common.naming_utils import (
+    generate_experiment_name,
+    generate_run_name,
+    generate_environment_name,
+)
+from mlops.common.pipeline_job_config import PipelineJobConfig
+
+
+def set_pipeline_properties(
+    pipeline_job: pipeline,
+    cluster_name: str,
+    display_name: str,
+    tags: dict,
+    default_datastore: str = "workspaceblobstore",
+    force_rerun: bool = True,
+):
+    """
+    Set properties for the pipeline job.
+
+    Args:
+        pipeline_job (pipeline): The pipeline job to set properties for.
+        cluster_name (str): The name of the compute cluster.
+        display_name (str): The display name for the pipeline job.
+        tags (dict): A dictionary of key-value pairs to set as tags for the pipeline job.
+        default_datastore (str, optional): The default datastore for the pipeline job. Defaults to "workspaceblobstore".
+        force_rerun (bool, optional): Whether to force rerun the pipeline job. Defaults to True.
+
+    Returns:
+        pipeline: The updated pipeline job with the specified properties.
+    """
+    pipeline_job.display_name = display_name
+    pipeline_job.tags = tags
+
+    # set pipeline level compute
+    pipeline_job.settings.default_compute = cluster_name
+    pipeline_job.settings.force_rerun = force_rerun
+    # set pipeline level datastore
+    pipeline_job.settings.default_datastore = default_datastore
+
+    return pipeline_job
+
+
+def execute_pipeline(
+    subscription_id: str,
+    resource_group_name: str,
+    workspace_name: str,
+    experiment_name: str,
+    pipeline_job: pipeline,
+    wait_for_completion: str,
+    output_file: str,
+):
+    """
+    Execute a pipeline job in Azure Machine Learning service.
+
+    Args:
+        subscription_id (str): The Azure subscription ID.
+        resource_group_name (str): The name of the resource group.
+        workspace_name (str): The name of the Azure Machine Learning workspace.
+        experiment_name (str): The name of the experiment.
+        pipeline_job (pipeline): The pipeline job to be executed.
+        wait_for_completion (str): "True" or "False" indicates whether to wait for the job to complete.
+        output_file (str): The path to the output file where the job name will be written.
+
+    Raises:
+        Exception: If the job fails to complete.
+    """
+    try:
+        client = MLClient(
+            DefaultAzureCredential(),
+            subscription_id=subscription_id,
+            resource_group_name=resource_group_name,
+            workspace_name=workspace_name,
+        )
+
+        pipeline_job = client.jobs.create_or_update(
+            pipeline_job, experiment_name=experiment_name
+        )
+
+        print(f"The job {pipeline_job.name} has been submitted!")
+        if output_file is not None:
+            with open(output_file, "w") as out_file:
+                out_file.write(pipeline_job.name)
+
+        if wait_for_completion == "True":
```

Comment:
how big if a refactor would it be to be a good citizen and change this to type bool in the codebase?

I ask because anywhere in the code where we read from argparse, we can store a value of type bool using the helper method below

```
def str2bool(value):
    if isinstance(value, bool):
        return value
    if value.lower() in ('yes', 'true', 't', 'y', '1'):
        return True
    elif value.lower() in ('no', 'false', 'f', 'n', '0'):
        return False
    else:
        raise argparse.ArgumentTypeError('Boolean value expected.')```

Reply:
I'm aligned with you and think there is some improvement we can do here. I tried to see how argparse does with boolean inputs and it doesn't do great 😅. However I did see that it can do command features so we could have use `--wait_for_completion` as is to have it wait for completion and not include that flag if we don't want to wait. We could also do a default action of `store_true` which when the flag is present set the variable to a boolean True. This would be a bit bigger of a refactor though and touch the ci/cd pipelines, but I think overall it might have advantages over converting a string to a boolean and checking the string is a match. The only hiccup I have currently is I'm not exactly sure how do accomplish this flag option in our VS Code tasks at the moment as an input selection-type thing. Thoughts on all of this? I think for this PR, I'm going to leave it as is since I'd like to explore this more and it's a bit out of scope. (Sorry this is long)